File: app/src/main/java/com/unionpay/numberpassword/NumberPassword/NumberPassword.java
package com.unionpay.numberpassword.NumberPassword;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.unionpay.numberpassword.R;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.UnsupportedEncodingException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.Random;

public class NumberPassword extends AppCompatActivity {

    Map<String, ArrayList<String>> number2obj = new HashMap<>();
    ArrayList<String> keys;
    String key;

    Button button;
    TextView textView1;
    TextView textView2;
    String string1, string2;
    String[] pair;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.number_password);

        InputStream inputStream = getResources().openRawResource(R.raw.numberpassword);
        getString(inputStream);
        pair = getPair();

        string1 = pair[0];
        string2 = pair[1];
        textView1 = (TextView) findViewById(R.id.TextView1);
        textView1.setText(string1);

        textView2 = (TextView) findViewById(R.id.TextView2);

        button = (Button) findViewById(R.id.button1);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                string2 = pair[1];
                textView2.setText(string2);
            }
        });

        button = (Button) findViewById(R.id.button2);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pair = getPair();
                string1 = pair[0];
                textView1.setText(string1);
                textView2.setText(null);
            }
        });


    }

    String[] getPair() {
        Random random = new Random();

        key = keys.get(random.nextInt(keys.size()));
        ArrayList<String> values = number2obj.get(key);
        String value = values.get(random.nextInt(values.size()));


        if (random.nextInt(2) == 0)
            return new String[]{key, value};
        else
            return new String[]{value, key};

    }


    String getString(String filepath) {
        File file = new File(filepath);
        FileInputStream fileInputStream = null;
        try {
            fileInputStream = new FileInputStream(file);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        return getString(fileInputStream);
    }

    String getString(InputStream inputStream) {
        InputStreamReader inputStreamReader = null;
        try {
            inputStreamReader = new InputStreamReader(inputStream, "utf-8");
        } catch (UnsupportedEncodingException e1) {
            e1.printStackTrace();
        }
        BufferedReader reader = new BufferedReader(inputStreamReader);
        StringBuffer sb = new StringBuffer("");
        String line;

        try {
            while ((line = reader.readLine()) != null) {
                sb.append(line);
                sb.append("\n");
                addToMap(line);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        keys = new ArrayList<>(number2obj.keySet());
        return sb.toString();
    }

    void addToMap(String line) {
        String[] items = line.split("、");
        ArrayList<String> objs = new ArrayList<>();
        if (items.length < 2)
            return;
        for (int i = 1; i < items.length; i++) {
            objs.add(items[i]);
        }
        number2obj.put(items[0], objs);
    }

}

File: app/src/main/java/com/unionpay/numberpassword/NumberPassword/NumberPassword.kt
package com.unionpay.numberpassword.NumberPassword

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.TextView
import com.unionpay.numberpassword.R
import java.util.*

class NumberPassword : AppCompatActivity() {

    internal var number2obj: MutableMap<String, List<String>> = HashMap()

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.number_password)

        getString()
        var pair = getPair()
//
        var textView1 = findViewById(R.id.TextView1) as TextView
        textView1.text = pair[0]
//
        var textView2 = findViewById(R.id.TextView2) as TextView

        var button = findViewById(R.id.button1) as Button
        button.setOnClickListener {
            textView2.text = pair[1]
        }
//
        button = findViewById(R.id.button2) as Button
        button.setOnClickListener {
            pair = getPair()
            textView1.text = pair[0]
            textView2.text = null
        }


    }

    fun getString() {

        var stream = resources.openRawResource(R.raw.numberpassword)
        var items = stream.reader().readLines()

        items.forEach { it ->
            var values = it.split("、")
            var key = values[0]
            var value = values.filter { values.indexOf(it) > 0 }
            number2obj[key] = value
        }
        println("Done!")
    }

    internal fun getPair(): List<String?> {
        val random = Random()

        val key = random.nextInt(number2obj.size).toString()
        val values = number2obj[key]
        val value = values?.get(random.nextInt(values.size))


        if (random.nextInt(2) == 0)
            return listOf(key, value)
        else
            return listOf(key, value)
    }

    internal fun addToMap(line: String) {
        val items = line.split("、".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val objs = ArrayList<String>()
        if (items.size < 2)
            return
        for (i in 1..items.size - 1) {
            objs.add(items[i])
        }
        number2obj.put(items[0], objs)
    }

}
